triggers/service/src/test/scala/com/digitalasset/daml/lf/engine/trigger/ServiceTest.scala

```diff
@@ -100,47 +109,97 @@ class ServiceTest extends AsyncFlatSpec with Eventually with Matchers {
     Http().singleRequest(req)
   }
 
+  def responseBodyToString(resp: HttpResponse) = {
+    resp.entity.dataBytes.runFold(ByteString(""))(_ ++ _).map(_.utf8String)
+  }
+
   it should "should fail for non-existent trigger" in withHttpService(Some(dar)) {
     (uri: Uri, client) =>
       for {
-        resp <- startTrigger(uri, s"${dar.main._1}:TestTrigger:foobar", "Alice")
+        resp <- startTrigger(uri, s"$testPkgId:TestTrigger:foobar", "Alice")
         body <- {
           assert(resp.status == StatusCodes.UnprocessableEntity)
-          resp.entity.dataBytes.runFold(ByteString(""))(_ ++ _).map(_.utf8String)
+          responseBodyToString(resp)
         }
       } yield assert(body == "Could not find name foobar in module TestTrigger")
   }
 
-  it should "find a trigger after uploading it" in withHttpService(None) { (uri: Uri, client) =>
-    for {
-      // attempt to start trigger before uploading which fails.
-      resp <- startTrigger(uri, s"${dar.main._1}:TestTrigger:trigger", "Alice")
-      _ <- assert(resp.status == StatusCodes.UnprocessableEntity)
-      resp <- uploadDar(uri, darPath)
-      body <- resp.entity.dataBytes.runFold(ByteString(""))(_ ++ _).map(_.utf8String)
-      _ <- body should startWith("DAR uploaded")
-      resp <- startTrigger(uri, s"${dar.main._1}:TestTrigger:trigger", "Alice")
-      _ <- assert(resp.status.isSuccess)
-      triggerId <- resp.entity.dataBytes.runFold(ByteString(""))(_ ++ _).map(_.utf8String)
-      resp <- stopTrigger(uri, triggerId)
-      _ <- assert(resp.status.isSuccess)
-    } yield succeed
+  it should "find a trigger after uploading it" in withHttpService(None) {
+    (uri: Uri, client) =>
+      for {
+        // attempt to start trigger before uploading which fails.
+        resp <- startTrigger(uri, s"$testPkgId:TestTrigger:trigger", "Alice")
+        _ <- assert(resp.status == StatusCodes.UnprocessableEntity)
+        resp <- uploadDar(uri, darPath)
+        body <- responseBodyToString(resp)
+        _ <- body should startWith("DAR uploaded")
+        resp <- startTrigger(uri, s"$testPkgId:TestTrigger:trigger", "Alice")
+        _ <- assert(resp.status.isSuccess)
+        triggerId <- responseBodyToString(resp)
+        resp <- listTriggers(uri, "Alice")
+        body <- responseBodyToString(resp)
+        _ <- body should include(triggerId)
+        resp <- stopTrigger(uri, triggerId)
+        _ <- assert(resp.status.isSuccess)
+      } yield succeed
+  }
+
+  it should "start multiple triggers and list them by party" in withHttpService(Some(dar)) {
+    (uri: Uri, client) =>
+      for {
+        // no triggers running initially
+        resp <- listTriggers(uri, "Alice")
+        _ <- assert(resp.status.isSuccess)
```

Comment:
It looks like some helper that throws on failed requests might come in handy but doesn’t need to be in this PR.